packages/sdk/src/swap/v2/types.ts

```diff
@@ -72,89 +65,52 @@ export namespace SwapV2 {
     latestSwapRescheduledAt: number | undefined;
     latestSwapRescheduledBlockIndex: number | undefined;
   };
-
   type SwapFields = {
     totalInputAmountSwapped: string | undefined;
     totalOutputAmountSwapped: string | undefined;
     totalChunksExecuted: number;
     isDcaSwap: boolean;
-    lastExecutedChunk: Swap | undefined;
-    currentChunk: Swap;
+    lastExecutedChunk: SwapChunk | undefined;
+    currentChunk: SwapChunk;
     type: SwapType;
     fees: SwapFee[];
   };
-
-  interface SwapStatusResponseCommonFields extends ChainsAndAssets {
+  type SwapStatusResponseCommonFields = ChainsAndAssets & {
     swapId: string;
     destAddress: string;
     depositChannel: DepositChannelFields;
     ccm: Ccm | undefined;
     boost: Boost | undefined;
     estimatedDurationSeconds: number | null | undefined;
-  }
-
+  };
   type EgressFields = {
     amount: string;
     scheduledAt: number;
     scheduledBlockIndex: string;
-  };
-  type EgressSentFields = EgressFields & {
-    sentTxRef: string;
-  };
-  type EgressCompletedFields = EgressFields & {
-    sentAt: number;
-    sentAtBlockIndex: string;
+    sentTxRef: string | undefined;
+    sentAt: number | undefined;
+    sentAtBlockIndex: string | undefined;
+    failure: Failure | undefined;
   };
   type RefundFields = {
     ignoredAmount: string | undefined;
   };
+  type SendingSwapFields = (SwapFields & { egress: EgressFields }) | undefined;
+  type SendingRefundFields = (RefundFields & EgressFields) | undefined;
 
-  type WithEgressFields<T, F = undefined> = T & {
-    egress: (F extends undefined ? EgressFields : EgressFields & { failure: F }) | undefined;
-  };
-  type WtihEgressSentFields<T, F = undefined> = T & {
-    egress:
-      | (F extends undefined ? EgressSentFields : EgressSentFields & { failure: F })
-      | undefined;
+  type DepositChannel = SwapStatusResponseCommonFields & {
+    depositChannel: DepositChannelFields;
   };
-  type WtihEgressCompletedFields<T> = T & {
-    egress: EgressCompletedFields | undefined;
+  type Receiving = DepositChannel & {
+    deposit: DepositFields;
   };
-
-  type SendingSwapFields = WithEgressFields<SwapFields, Failure> | undefined;
-  type SentSwapFields = WtihEgressSentFields<SwapFields, Failure> | undefined;
-  type CompletedSwapFields = WtihEgressCompletedFields<SwapFields> | undefined;
-
-  type SendingRefundFields = WithFailure<RefundFields & EgressFields> | undefined;
-  type SentRefundFields = WithFailure<RefundFields & EgressSentFields> | undefined;
-  type CompletedRefundFields = EgressCompletedFields | undefined;
-
-  interface DepositChannel extends SwapStatusResponseCommonFields {
-    depositChannel: DepositChannelFields;
-  }
-  interface Receiving extends DepositChannel {
-    deposit: WithFailure<DepositFields>;
-  }
-  interface Swapping extends Receiving {
+  type Swapping = Receiving & {
     swap: SwapFields;
-  }
-  interface Sending extends Receiving {
```

Comment:
maybe extending `interface`s provides a better DX? usually type intersections provide some janky, hard to read types